modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/actions/PreIssueAccessTokenPasswordGrantTestCase.java

```diff
@@ -122,24 +129,47 @@ public class PreIssueAccessTokenPasswordGrantTestCase extends ActionsBaseTestCas
     private static final String NEW_SCOPE_4 = "replaced_scope";
 
     private static final String SCIM2_USERS_API = "/o/scim2/Users";
-    private static final String ACTIONS_API = "/api/server/v1/actions";
-    private static final String APPLICATION_MANAGEMENT_API = "/api/server/v1/applications";
-    private static final String API_RESOURCE_MANAGEMENT_API = "/api/server/v1/api-resources";
-    private static final String MOCK_SERVER_ENDPOINT = "/test/action";
-    protected SCIM2RestClient scim2RestClient;
+    private static final String CLAIMS_PATH_PREFIX = "/accessToken/claims/";
+    private static final String SCOPES_PATH_PREFIX = "/accessToken/scopes/";
+    private static final String MOCK_SERVER_ENDPOINT_RESOURCE_PATH = "/test/action";
+    private static final int UPDATED_EXPIRY_TIME_PERIOD = 7200;
+    private static final int CURRENT_EXPIRY_TIME_PERIOD = 3600;
+    private SCIM2RestClient scim2RestClient;
     private Lookup<CookieSpecProvider> cookieSpecRegistry;
     private RequestConfig requestConfig;
     private CloseableHttpClient client;
     private List<String> customScopes;
+    private List<String> requestedScopes;
     private String accessToken;
     private String clientId;
     private String clientSecret;
+    private String subjectType;
+    private String tokenType;
     private String actionId;
     private String applicationId;
     private String domainAPIId;
     private String userId;
     private String roleId;
+    private String tenantId;
     private JWTClaimsSet jwtClaims;
+    private TestUserMode userMode;
+    private ActionsMockServer actionsMockServer;
+
+    @Factory(dataProvider = "testExecutionContextProvider")
+    public PreIssueAccessTokenPasswordGrantTestCase(TestUserMode testUserMode) {
+
+        this.userMode = testUserMode;
+        this.tenantId = testUserMode == TestUserMode.SUPER_TENANT_USER ? "-1234" : "1";
+    }
+
+    @DataProvider(name = "testExecutionContextProvider", parallel = true)
```

Comment:
Shall we move this to ActionBaseTestCase class and add the dataprovider to the the ClientCredentials test class as well ?